exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/QueryManager.java

```diff
@@ -287,6 +288,7 @@ QueryState updateEphemeralState(final QueryState queryState) {
       case CANCELED:
       case FAILED:
         try {
+          profilePStore.put(stringQueryId, getQueryProfile());
```

Comment:
Should be removed. Foreman.ForemanResult#close has the call to write the final profile.